### ISSUE TITLE
False negative: $this in nested function declarations

### ISSUE THREAD

**Issue by jrfnl** [sirbrillig/phpcs-variable-analysis#109]
```php
$closure = function() {
    // Using $this here is fine.
    if ($this->something) {
        function nestedFunctionDeclaration() {
            // Using $this here is not as the nested function is a closed scope in the global namespace.
            if ($this->something) {
                // Do something.
            }
        }
    }
};
```

The use of `$this` within the `nestedFunctionDeclaration()` function is not reported.

I have a fix ready for this, but need a decision on #108 before I can pull it.
